### DISCUSSION TITLE
Testing introduction

### DISCUSSION THREAD

**Discussion by groteck** [nuxt/nuxt#17708] [Questions]
Hi all, I was looking into Vue3 docs and there is nothing related to testing. I understand that is a Beta version and the docs are in a early stage. But I think will be cool to add a testing section about how to test some stuff.

**Comment by lucpotage** [comment]
Since Nuxt 3 will support Vite and Webpack 5, it takes quite some time to offer a "bundler-agnostic" experience. In case you missed it, @antfu works on [Vitest](https://vitest.dev/), a unit-test framework powered by Vite. It looks super promising, but it's clearly not ready yet (close beta for Sponsosrs).

**Reply by joel-wenzel** [reply]
I have vitest running with nuxt 3 but there are two main issues I see:

- Vitest uses the same `vite.config.ts` file found in the project root. Nuxt generates the vite.config dynamically so vitest needs some way to inject that config
- Auto imports - I was able to brute force my way through the vite.config but still couldn't get auto imports working correctly.

From my initial experience vitest is super easy to use so it would be awesome if it worked with nuxt 3.

**Reply by itsmnthn** [reply]
I am having same issues you mentioned. Mostly not working is relative import inside file that I am testing

**Reply by itsmnthn** [reply]
Please do share if you have any work around

**Comment by lucpotage** [comment]
There is also [peeky.dev](https://peeky.dev/) from Guillaume Chau.
